### PR TITLE
Docker implementation

### PR DESCRIPTION
**Pull request by chriswedgwood**
This is to make testing production like scenarios with elastic search and postgres easier.
I have put most of the third party version updates into this pull request rather than doing them one by one.
This is my first stab at implementing docker so would be very happy to hear any recommendations on its setup.
I have not upgraded elastic search in this pull request as I had struggles getting it working on a later version and I am not convinced we need to upgrade it. 

If accepted you can close:
#110 
#109 
#108(I removed pyelasticsearch ERROR: pyelasticsearch 1.4.1 has requirement elasticsearch<2.0.0,>=1.3.0)
#94 
#105 
#104 
#103(not sure about upgrading this one may need to revert and  add gunicorn to docker setup)
#94 

It would be useful to know what versions we are running in production?

### CODE REVIEW

**Comment by carltongibson** [comment]
Hi @chriswedgwood   This looks fun. I shall get to it later in the week. Thanks!

**Comment by chriswedgwood** [comment]
> Hi @chriswedgwood.
> 
> This is great. Can I ask you to just split the docker changes from the requirements changes (so we merge just the Docker stuff, then update the dependencies)?

Will do thanks @carltongibson

**Comment by chriswedgwood** [comment]
@carltongibson that's done thanks

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      - esdata:/usr/share/elasticsearch/data
+    ports:
+      - "9200:9200"
+      - "9300:9300"
+volumes:
+  redisdata:
+  postgresdata:
+  esdata:
+    driver: local
+
```
Comment (carltongibson) [inline docker-compose.yml:56]: Can you trim whitespace? I think this No newline will vanish then.

Comment (chriswedgwood) [reply docker-compose.yml]: Fixed with insertFinalNewline in vscode

**Pull request merged**